### DISCUSSION TITLE
How should I get fluidsynth to ignore tempo information

### DISCUSSION THREAD

**Discussion by nolanzurek** [FluidSynth/fluidsynth#1103] [Q&A] [answered]
I am currently writing a MIDI visualizer in processing that uses fluidsynth to generate the audio for the visualization (my processing script then combines it with the video output). My visualizer does not take tempo change events in the MIDI file into account; it simply sets the whole visualization to the initial tempo of the file (ignoring all subsequent tempo change events). However, when tempo changes exist in the file, fluidsynth "renders" them as one would expect, leading to the visualization and audio becoming unsynchronized.

Is there a way to do either of the following things when calling fluidsynth from the command line

- Have it ignore subsequent tempo markings in the MIDI file and "render" the audio at the initial tempo
- Force it to "render" the audio at a given tempo (presumably taken as a command line argument)

**Comment by jimhen3ry** [comment, marked as answer]
The FluidSynth API document says that fluid_player_set_tempo() can override the tempo provided by a MIDI file: https://www.fluidsynth.org/api/group__midi__player.html#gab99758ae72fc14dd149a64cb3520700a
